Update Unifi bandwidth sensors

wittypluck opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

This Pull request aim is to improve Unifi network bandwidth sensor. Currently, bandwidth sensors (RX, TX) have the unit set as MEGABYTE, and are not flagged as Measurement. They need to be wrapped in template sensors to be used as bandwidth sensors.

Updating the unit of measurement to MEGABYTES_PER_SECOND and the device class to DATA_RATE ensure that the sensors can be used "as is" in Home assistant without adding additional template sensors to convert to the correct units.
Additionally the sensors are now automatically picked up by the History feature of Home assistant to provide long term statistics.
The PR also adds ~~a suggested unit of measurement (MEGABITS_PER_SECOND) and~~ an icon to each of the TX, RX sensors.

I have tested locally that the PR does not provoke regressions for existing users of the integration, and that changes to the sensors are picked up correctly for existing and new users of the integration.

I have also tested locally through bandwidth tests that although the Unifi documentation of the underlying sensors (rx_bytes-r / tx_bytes-r / wired-rx_bytes-r / wired-tx_bytes-r) is not very explicit, they do indeed correspond to a measure in bytes over the last second.

I have also tested locally that somewhat un-intuitively the "TX" sensor corresponds to download bandwidth (as seen from the device) and the "RX" sensor corresponds to upload bandwidth (as seen from the device).

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

wittypluck committed:
Change bandwidth sensors device class, state class, unit of measurement, icon

Kane610 commented:
Thanks for the PR. Please start by fixing the failing checks which seems to be related to unsorted imports.

wittypluck committed:
Reformat imports

wittypluck committed:
Merge branch 'dev' into dev

wittypluck committed:
Reformat imports

wittypluck committed:
Merge branch 'dev' into dev

wittypluck commented:
Thank you for the comment and the tip, I'm still having issues with the pre commit tools but will make sure the formatting is correct.

wittypluck committed:
Merge branch 'dev' into dev

Kane610 reviewed (approved):
Thanks

wittypluck committed:
Merge branch 'dev' into dev

wittypluck commented:
Hi @Kane610, thanks for approving the change. It seems I still have a few tests to fix, I'm working on it.

Kane610 commented:
> Hi @Kane610, thanks for approving the change. It seems I still have a few tests to fix, I'm working on it.

Apologies for loosing track of the review. Let's get this in before next beta branches off

wittypluck committed:
Revert suggested_unit_of_measurement. Add unit tests.

wittypluck commented:
No problem!
I have reverted the suggested_unit_of_measurement which was converting from MEGABYTES_PER_SECOND to MEGABITS_PER_SECOND as it was causing issues with unit tests and could be an unwanted change for some users.
I have also updated and added unit tests on the new attributes so hopefully everything should pass now.

Kane610 commented:
Thank you @wittypluck

wittypluck commented:
Thank you @Kane610 as this was my first commit in home assistant !

Kane610 commented:
Welcome to the club @wittypluck
